### PR TITLE
fix: small changes to E2E tests to use header for passing service ID

### PR DESCRIPTION
**Pull request by commoddity**
Updates to remove deprecated usage of `subdomain` for passing service ID in favour of passing it in the config.

Also cleans up the example config files that are used in the E2E tests to remove references to `alias`.

### CODE REVIEW

File: e2e/morse_relay_test.go [hunk 1]
```diff
@@ -24,26 +25,28 @@ func Test_MorseRelay(t *testing.T) {
...
 		{
-			name:         "should successfully relay eth_chainId for eth (F00C)",
-			reqMethod:    http.MethodPost,
-			serviceAlias: "F00C",
-			relayID:      "1201",
-			body:         `{"jsonrpc": "2.0", "id": "1201", "method": "eth_chainId"}`,
+			name:      "should successfully relay eth_chainId for eth (F00C)",
+			reqMethod: http.MethodPost,
+			serviceID: "F00C",
+			reqPath:   "/v1/abcdef12",
```
Comment (adshmh) [inline e2e/morse_relay_test.go:39]: This should not be made part of the test case, as it is the same for all test cases.

Comment (commoddity) [reply e2e/morse_relay_test.go:39]: Yes but we may want the functionality to run the test for other services, which is why I added it to the test variables.

File: e2e/shannon_relay_test.go [hunk 2]
```diff
@@ -37,27 +39,28 @@ func Test_ShannonRelay(t *testing.T) {
 			// single endpoint, maintained by Grove.
 			name:      "should successfully relay eth_blockNumber for anvil",
 			reqMethod: http.MethodPost,
+			serviceID: "anvil",
+			reqPath:   "/v1/abcdef12",
```
Comment (adshmh) [inline e2e/shannon_relay_test.go:43]: Ditto here: `reqPath` is the same for all the test cases. If at some point that changes, we can update the tests accordingly.

File: e2e/shannon_relay_test.go [hunk 3]
```diff
@@ -37,27 +39,28 @@ func Test_ShannonRelay(t *testing.T) {
...
 			relayID:   "1002",
 			body:      `{"jsonrpc": "2.0", "id": "1002", "method": "eth_chainId"}`,
 		},
 		// TODO_UPNEXT(@adshmh): add more test cases with valid and invalid jsonrpc request payloads.
 	}
 
-	reqPath := "/v1/abcdef12"
-	serviceID := "anvil"
```
Comment (adshmh) [inline e2e/shannon_relay_test.go:53]: I don't think this should be moved to individual tests: `anvil` is going to be used for Shannon E2E tests for now.

**Review by adshmh — CHANGES_REQUESTED** [review]
Thank you for the PR. Just a few minor blockers to address.

File: e2e/shannon_relay_test.go (hunk 2 above)
Comment (commoddity) [reply e2e/shannon_relay_test.go:43]: > Ditto here: `reqPath` is the same for all the test cases. If at some point that changes, we can update the tests accordingly.

I take your point but curious what the harm is in having them as test variables vs singlularly defined vars? My preference is generally to have any variables used in the test defined in the test cases for both consistency and the possibility of using other values down the line.

File: e2e/morse_relay_test.go (hunk 1 above)
Comment (adshmh) [reply e2e/morse_relay_test.go:39]: PATH does not differentiate between requests based on this part of the URL at this point. E2E tests need to make this clear by explicitly declaring the value as a single variable/constant used by all the tests.

Comment (commoddity) [reply e2e/morse_relay_test.go:39]: OK for `reqPath` you're right and that makes sense. But for the service ID IMO it's worth keeping as a test far.

File: e2e/shannon_relay_test.go (hunk 2 above)
Comment (adshmh) [reply e2e/shannon_relay_test.go:43]: request path is not really `variable` here: all the tests use the same value. Declaring it as such makes it a lot easier to read the E2E tests by anyone who is not familiar with how PATH treats different parts of the request URL.

Comment (commoddity) [reply e2e/shannon_relay_test.go:43]: > request path is not really `variable` here: all the tests use the same value. Declaring it as such makes it a lot easier to read the E2E tests by anyone who is not familiar with how PATH treats different parts of the request URL.

Got it; I do agree with this re: the request path.

**Review by adshmh — APPROVED** [review]

**Pull request merged**